src/main/java/com/ss/training/lms/controller/AdminGenreController.java

```diff
@@ -0,0 +1,123 @@
+package com.ss.training.lms.controller;
+
+import java.sql.SQLException;
+import java.util.List;
+
+import com.ss.training.lms.entity.Genre;
+import com.ss.training.lms.service.admin.AdminGenreService;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * @author Trevor Huis in 't Veld
+ */
+@RestController
+public class AdminGenreController {
+	
+	@Autowired
+	AdminGenreService genreService;
+	
+	@RequestMapping(path="/lms/admin/genre/{genreId}", 
+					produces = {
+						MediaType.APPLICATION_XML_VALUE,
+						MediaType.APPLICATION_JSON_VALUE
+					})
+	public ResponseEntity<Genre>getGenreById(@PathVariable int genreId) {
+		Genre genre = null;
+		HttpStatus status = HttpStatus.OK;
+		try {
+			genre= genreService.readGenre(genreId);
+			if (genre == null) {
+				status = HttpStatus.NOT_FOUND;
+			}
+		} catch (SQLException e) {
+			status = HttpStatus.INTERNAL_SERVER_ERROR;
+		}
+		return new ResponseEntity<Genre>(genre, status);
+	}
+
+	@RequestMapping(path="/lms/admin/genres", 
+					produces = {
+						MediaType.APPLICATION_XML_VALUE,
+						MediaType.APPLICATION_JSON_VALUE
+					})
+	public ResponseEntity<List<Genre>> getAllGenres() {
+		List<Genre> genres = null;
+		HttpStatus status = HttpStatus.OK;
+        try {
+			genres= genreService.readAllGenres();
+			if (genres == null) {
+				status = HttpStatus.NOT_FOUND;
```

Comment:
Have you considered setting the status to No Content here? I was thinking about it, and I kind of feel like it makes sense: The client requested all records from a given table. We have the table but it has no records, so we basically tell them we successfully processed the request and the result is no content. Not Found, on the other hand, would seem to suggest the table doesn't exist. Just food for thought.